### PR TITLE
pythonPackages.progressbar2: init at 3.12.0

### PR DESCRIPTION
**Pull request by ashgillman**
###### Motivation for this change
New python package, simple to use progress bar

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/progressbar2/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  name = "${pname}-${version}";
+  pname = "progressbar2";
+  version = "3.12.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "16r21cpjvv0spf4mymgpy7hx6977iy11k44n2w9kipwg4lhwh02k";
+  };
+
+  buildInputs = [ pytest ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/progressbar2/default.nix]: `pytest` belongs into `checkInputs`.

**Review by dotlambda — CHANGES_REQUESTED** [review]
Why did you disable tests?

`setup.py` seems to download missing requirements:
```
Download error on https://pypi.python.org/simple/pytest-runner/: [Errno -2] Name or service not known -- Some packages may not be found!
Download error on https://pypi.python.org/simple/: [Errno -2] Name or service not known -- Some packages may not be found!
```
But no luck even if we use `pytestrunner` from nixpkgs, since `pytest-runner>=2.8` is required. We will need to update it beforhand.

Maybe you didn't see the download errors if you don't have enabled sandboxing.

**Comment by ashgillman** [comment]
Last commit should have dealt with all issues. I had a little trouble getting tests to run, and currently need to pull from GitHub, not PyPI (notified upstream, commented in expression).

**Comment by dotlambda** [comment]
@FRidh Shouldn't have been merged. I still get
```
Download error on https://pypi.python.org/simple/pytest-runner/: [Errno -2] Name or service not known -- Some packages may not be found!
```
We need to add `pytestrunner` as a dependency.

**Comment by FRidh** [comment]
Right, I missed that. 

It seems `pytestrunner` needs to be split out and updated.

**Pull request merged**